source/simulator/solver_schemes.cc

```diff
@@ -1452,6 +1455,149 @@ namespace aspect
     nonlinear_solver_control.check(1,0.0);
     signals.post_nonlinear_solver(nonlinear_solver_control);
   }
+
+
+
+  template <int dim>
+  void Simulator<dim>::solve_stokes_adjoint ()
+  {
+    // TODO: set an assert here or elsewhere that the material model additive is needed if the max_nonlinear_iteration is more than 1
+    // What is the best way of getting the material model name?
```

Comment:
That's typically done by doing something along the lines of 
```
  Assert (dynamic_cast<const Additive<dim>*>(&get_material_model()) != nullptr,
```
that is, you don't ask for the symbolic name but just see whether you can cast the material model object to the desired type.